### DISCUSSION TITLE
Cannot read RLE Lossless DICOM images

### DISCUSSION THREAD

**Discussion by jochalek** [ImageMagick/ImageMagick#6669] [Contribute]
### ImageMagick version

7.1.1-15

### Operating system

Linux

### Operating system, version and so on

Ubuntu 22.04.3 LTS

### Description

Attempting to read RLE Lossless DICOM images with Transfer Syntax UID "1.2.840.10008.1.2.5" results in ```improper image header``` error.

I'm not certain what the problem is. My DICOM data are RLE Lossless encoded and cause this error. Any RLE lossless encoded samples I can find elsewhere also cause this error. DICOMs encoded in other formats do work. 

I can reproduce the same problem with ImageMagick6 as well.

### Steps to Reproduce

```
user@ubuntu:~/$ identify ~/Downloads/US-PAL-8-10x-echo
identify: improper image header `/home/user/Downloads/US-PAL-8-10x-echo' @ error/dcm.c/ReadDCMImage/3210.
```

### Images

The example image may be found here [US-PAL-8-10x-echo](https://www.dropbox.com/s/gkl0jsv41olq9n1/US-PAL-8-10x-echo.gz?dl=0).

**Comment by dlemstra** [comment]
It looks like this image contains something we don't support or we are reading the RLE compression incorrectly. Someone will need to investigate that.

Adding the file here for future reference: [US-PAL-8-10x-echo.gz](https://github.com/ImageMagick/ImageMagick/files/12365621/US-PAL-8-10x-echo.gz)

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce the issue and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.
